### ISSUE TITLE
Running private-ghp with localhost

### ISSUE THREAD

**Issue by franyol** [john-dev/private-ghp#1]
I'm testing the code running it in local host for start, setting the OAUTH app's callback URL pointing to http://test.localhost:8080/login/github/callback

Also I'm building using the following config.yaml:

```
port: 8080
publicPort: 8080
domain: localhost
log:
  level: debug
github:
  client:
    id: (UserID)
    secret:  (UserSecret)
pages:
  - subdomain: test #the sub-domain part WITHOUT the main domain
    index: index.html
    cache:
      duration: 1800   
    repository:
      owner: kiwicampus
      name: kronos-project-docs
      branch: privpage # eg. gh-pages
```

After building it, entering test.localhost:8080 and clicking the "Authorize" button; I get redirected to the same authorization path over and over.

This is the terminal I get by doing so:

```
fran@fran:~/Desktop/GITREP/private-ghp$ ./bin/linux/private-ghp --config=sample.config.yaml
DEBU[2022-04-28 13:05:33] loaded config file: sample.config.yaml       
INFO[2022-04-28 13:05:33] listening on 0.0.0.0:8080                    
DEBU[2022-04-28 14:09:41] token recevied from github, redirecting to http://test.localhost:8080// 
DEBU[2022-04-28 14:09:42] token recevied from github, redirecting to http://test.localhost:8080// 
DEBU[2022-04-28 14:09:43] token recevied from github, redirecting to http://test.localhost:8080// 
DEBU[2022-04-28 14:09:43] token recevied from github, redirecting to http://test.localhost:8080// 
DEBU[2022-04-28 14:09:44] token recevied from github, redirecting to http://test.localhost:8080// 
DEBU[2022-04-28 14:09:46] token recevied from github, redirecting to http://test.localhost:8080// 
DEBU[2022-04-28 14:09:46] token recevied from github, redirecting to http://test.localhost:8080// 
DEBU[2022-04-28 14:09:47] token recevied from github, redirecting to http://test.localhost:8080// 
DEBU[2022-04-28 14:09:48] token recevied from github, redirecting to http://test.localhost:8080// 
```

It seems I'm getting into some kind of loop, can you tell me what I'm doing wrong?

**Comment by YoSev** [comment]
Hey,

the callback should not go back to the subdomain but the domain, in your case http://localhost:8080/login/github/callback.
That's what allows you to add as many subdomains as you want with one instance of the service.

**Comment by franyol** [comment]
> Hey,
> 
> the callback should not go back to the subdomain but the domain, in your case http://localhost:8080/login/github/callback. That's what allows you to add as many subdomains as you want with one instance of the service.

You are right, the callback is actually pointing to http://localhost:8080/login/github/callback but I still get the same behaviour.

**Comment by YoSev** [comment]
The problem is localhost. 
TLDR;
only FQDN's are supported

I store cookies with the domain being set as ".domain". The prefixed dot tells the browser to send the cookie to subdomain requests too which is not working for localhost as cross origin resource sharing does not apply to it.
If you switch to something like local.host things should work fine.

**Comment by franyol** [comment]
I changed the localhost domain name as local.host as you said and now it's working, thanks a lot! 😃
